sdk/ai/azure-ai-inference/tests/test_model_inference_client.py

```diff
@@ -17,7 +21,281 @@ class TestModelClient(ModelClientTestBase):
 
     # **********************************************************************************
     #
-    #                      HAPPY PATH TESTS - TEXT EMBEDDINGS
+    #                               UNIT TESTS
+    #
+    # **********************************************************************************
+
+    def test_image_url_load(self, **kwargs):
+        local_folder = os.path.dirname(os.path.abspath(__file__))
+        image_file = os.path.join(local_folder, "test_image1.png")
+        image_url = sdk.models.ImageUrl.load(
+            image_file=image_file,
+            image_format="png",
+            detail=sdk.models.ImageDetailLevel.AUTO,
+        )
+        assert image_url
+        assert image_url.url.startswith("data:image/png;base64,iVBORw")
+        assert image_url.detail == sdk.models.ImageDetailLevel.AUTO
+
+    def test_chat_completions_with_defaults(self, **kwargs):
+        # pylint: disable=protected-access
+        client1 = sdk.ChatCompletionsClient(
+            endpoint="http://does.not.exist",
+            credential=AzureKeyCredential("key-value"),
+            some_input_arg="some_input_value",
+        )
+
+        client2 = client1.with_defaults(
+            model_extras={"key1": 1},
+            model="some-model-id",
+            frequency_penalty=0.123,
+            max_tokens=321,
+            presence_penalty=4.567,
+            response_format=sdk.models.ChatCompletionsResponseFormat.JSON_OBJECT,
+            seed=654,
+            stop=["stop1", "stop2"],
+            temperature=8.976,
+            tool_choice=sdk.models.ChatCompletionsToolSelectionPreset.AUTO,
+            tools=[ModelClientTestBase.TOOL1, ModelClientTestBase.TOOL2],
+            top_p=9.876,
+        )
+        assert client2 is not client1
+        assert client2._model_extras == {"key1": 1}
+        assert client2._model == "some-model-id"
+        assert client2._frequency_penalty == 0.123
+        assert client2._max_tokens == 321
+        assert client2._presence_penalty == 4.567
+        assert client2._response_format == sdk.models.ChatCompletionsResponseFormat.JSON_OBJECT
+        assert client2._seed == 654
+        assert client2._stop == ["stop1", "stop2"]
+        assert client2._temperature == 8.976
+        assert client2._tool_choice == sdk.models.ChatCompletionsToolSelectionPreset.AUTO
+        assert client2._tools == [ModelClientTestBase.TOOL1, ModelClientTestBase.TOOL2]
+        assert client2._init_kwargs.get("some_input_arg") == "some_input_value"
```

Comment:
One additional thing we should test is that the "child" client does not close the transport of the parent client. 

Here's an example test we have in another library: https://github.com/Azure/azure-sdk-for-python/blob/main/sdk/formrecognizer/azure-ai-formrecognizer/tests/test_ftc_mgmt_async.py#L123-L133

Reply:
Thanks @kristapratico I added these sync and async tests for the clients:

- test_async_embeddings_with_defaults_check_transport_not_closed
- test_async_chat_completions_with_defaults_check_transport_not_closed
- test_embeddings_with_defaults_check_transport_not_closed
- test_chat_completions_with_defaults_check_transport_not_closed

 Please have a look. There is some interesting observation in the new async tests, search for "# assert transport.session is None # Why is it not None here?". Any idea why the session is not None right after I created the async client (and before any service call is made)? It is None after I create the sync client.

Reply:
In your async test, you've used a context manager which calls __enter__, creating the session. In your sync test, you did not use a context manager. It looks like the tests are missing the crucial piece of calling an API from the parent client, after we've left the child's context manager (btw this is referring to the same issue that Johan mentioned here: https://github.com/Azure/azure-sdk-for-python/pull/36501#discussion_r1690544229)

Reply:
@kristapratico resolving this since we decided not to ship a `with_defaults` method, but instead add chat completions/embedding settings in the constructor.